bokehjs/src/lib/models/formatters/datetime_tick_formatter.ts

```diff
@@ -196,9 +194,10 @@ export class DatetimeTickFormatter extends TickFormatter {
       s = _strftime(t, this[next_resolution])
     }
 
-    if (this.strip_leading_zeros) {
+    const final_resolution = next_resolution
+    if ((typeof this.strip_leading_zeros === 'boolean' && this.strip_leading_zeros) || (Array.isArray(this.strip_leading_zeros) && this.strip_leading_zeros.includes(final_resolution))) {
```

Comment:
```suggestion
    const {strip_leading_zeros} = this
    if ((isBoolean(strip_leading_zeros) && strip_leading_zeros) || 
        (isArray(strip_leading_zeros) && strip_leading_zeros.includes(final_resolution))) {
```
`isBoolean` and `isArray` come from `core/util/types` module.